crates/viewer/re_viewer/src/app.rs

```diff
@@ -779,6 +772,176 @@ impl App {
         }
     }
 
+    /// Loads a data source into the viewer.
+    ///
+    /// Tries to detect whether the datasource is already present (either still streaming in or already loaded),
+    /// and if so, will not load the data again.
+    /// Instead, it will only perform any kind of selection/mode-switching operations associated with loading the given data source.
+    ///
+    /// Note that we *do not* change the display mode here _unconditionally_.
+    /// For instance if the datasource is a blueprint for a dataset that may be loaded later,
+    /// we don't want to switch out to it while the user browses a server.
+    fn load_datasource(
+        &mut self,
+        store_hub: &mut StoreHub,
+        egui_ctx: &egui::Context,
+        data_source: &DataSource,
+    ) {
+        // Check if we've already loaded this data source and should just switch to it.
+        //
+        // Go through all sources that are still loading and those that are already in the store_hub.
+        // (if we look only at the one from the store_hub, we might miss those that haven't hit it yet)
+        let active_sources = self.rx_log.sources();
+        let store_sources = store_hub
+            .store_bundle()
+            .entity_dbs()
+            .filter_map(|db| db.data_source.as_ref());
+        let mut all_sources = store_sources.chain(active_sources.iter().map(|s| s.as_ref()));
+
+        match data_source {
+            DataSource::RrdHttpUrl { url, follow } => {
+                let new_source = SmartChannelSource::RrdHttpStream {
+                    url: url.clone(),
+                    follow: *follow,
+                };
+                if all_sources.any(|source| source.is_same_ignoring_uri_fragments(&new_source)) {
+                    if let Some(entity_db) = store_hub.find_recording_store_by_source(&new_source) {
+                        if *follow {
+                            let rec_cfg = self.state.recording_config_mut(entity_db);
+                            let time_ctrl = rec_cfg.time_ctrl.get_mut();
+                            time_ctrl.set_play_state(
+                                entity_db.times_per_timeline(),
+                                PlayState::Following,
+                            );
+                        }
+
+                        let store_id = entity_db.store_id().clone();
+                        drop(all_sources);
+                        self.make_store_active_and_highlight(store_hub, egui_ctx, &store_id);
```

Comment:
Technically this could be a blueprint right? Should it then have the same `if store_id.is_recording()` check as the `DataSource::FilePath` branch below?